src/main/java/edu/harvard/iq/dataverse/TermsOfUseAndAccessValidator.java

```diff
@@ -12,21 +12,43 @@
  *
  * @author skraffmi
  */
-public class TermsOfUseAndAccessValidator implements ConstraintValidator<ValidateTermsOfUseAndAccess, TermsOfUseAndAccess>  {
+public class TermsOfUseAndAccessValidator implements ConstraintValidator<ValidateTermsOfUseAndAccess, TermsOfUseAndAccess> {
 
     @Override
     public void initialize(ValidateTermsOfUseAndAccess constraintAnnotation) {
-        
+
     }
 
     @Override
     public boolean isValid(TermsOfUseAndAccess value, ConstraintValidatorContext context) {
-        //if both null invalid
-        //if(value.getTemplate() == null && value.getDatasetVersion() == null) return false;
         
-        //if both not null invalid
-        //return !(value.getTemplate() != null && value.getDatasetVersion() != null);
-        return true;
+        return isTOUAValid(value, context);
+
+    }
+    
+    public static boolean isTOUAValid(TermsOfUseAndAccess value, ConstraintValidatorContext context){
+        //If there are no restricted files then terms are valid 
+        if (!value.getDatasetVersion().isHasRestrictedFile()) {
+            return true;
+        }
+        /*If there are restricted files then the version
+        must allow access requests or have terms of access filled in.
+         */
+        boolean valid = value.isFileAccessRequest() == true || (value.getTermsOfAccess() != null && !value.getTermsOfAccess().isEmpty());
+        if (!valid) {
+            try {
+                if (context != null) {
+                    context.buildConstraintViolationWithTemplate("If Request Access is false then Terms of Access must be provided.").addConstraintViolation();
+                }
+
+                String message = "Constraint violation found in Terms of Use and Access. "
+                        + " If Request Access to restricted files is set to false then Terms of Access must be provided.";
```

Comment:
This English should probably go in Bundle.properties so it can be translated.

Reply:
Ok. these have been moved to the Bundle. Possibly moot because the save button is disabled until TOA are in compliance.